pkg/tfgen/docs_test.go

```diff
@@ -846,6 +853,51 @@ func TestFindRepoPath(t *testing.T) {
 	assert.Equal(t, "./terraform-provider-bar", actual)
 }
 
+func TestExampleGeneration(t *testing.T) {
+	info := testprovider.ProviderMiniRandom()
+
+	markdown := []byte(`
+## Examples
+
+There is some more code in here.
+
+~~~java
+throw new Exception("!");
+~~~
+`)
+
+	markdown = bytes.ReplaceAll([]byte(markdown), []byte("~~~"), []byte("```"))
+
+	info.Resources["random_integer"].Docs = &tfbridge.DocInfo{
+		Markdown: markdown,
+	}
+
+	inmem := afero.NewMemMapFs()
+
+	g, err := NewGenerator(GeneratorOptions{
+		Package:      info.Name,
+		Version:      info.Version,
+		Language:     Schema,
+		ProviderInfo: info,
+		Root:         inmem,
```

Comment:
Using afero will write "pseudo files" in-memory instead of spilling to disk. What we expect is schema.json file is produced.